adafruit_minimqtt/adafruit_minimqtt.py

```diff
@@ -379,17 +409,20 @@ def connect(self, clean_session=True):
                 return result
 
     def disconnect(self):
-        """Disconnects the MiniMQTT client from the MQTT broker.
-        """
+        """Disconnects the MiniMQTT client from the MQTT broker."""
         self.is_connected()
         if self.logger is not None:
             self.logger.debug("Sending DISCONNECT packet to broker")
-        self._sock.send(MQTT_DISCONNECT)
+        try:
```

Comment:
This try/retry does not make sense to me - Why would we not be able to send a DISCONNECT control packet?

Disconnect packets (https://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc398718094) expect no response payload and should only close the connection with the server (and in our case, empty a list of prvsly subscribed topics)

Reply:
That is because sock.write() invoked by disconnect may fail. That happens if tcp session is in a bad state or wifi is down.

Reply:
@brentru if you would like, I can undo this change on this PR and open a separate issue for specifically dealing with the disconnect() / connect() bug. LMK, pls.

Reply:
I understand what you're trying to do here and agree that `socket.send` could fail if there was a network communication issue. Let's keep it in this PR 👍